src/main/java/com/google/sps/servlets/LoadTasksServlet.java

```diff
@@ -41,11 +40,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(tasksJson);
   }
 
-  private String getTasksJson() throws JsonProcessingException {
-    TasksProvider tasksProvider = new TasksProvider();
-    List<Task> tasks = tasksProvider.getTasks();
+  private String getTasksJson() throws IOException {
```

Comment:
I recommend being as precise as possible with exceptions. Is this really throwing an IOException now and not a more specific JsonProcessingException?

Reply:
That's because the instantiation of `TasksClientAdapter` throws an IOException, but I can move that in the `doGet` and pass it as a parameter.